charts/vmclarity/templates/apiserver/service.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: v1
+kind: Service
```

Comment:
Not sure why we need both service and headless-service for the api-server. Can you please explain?

Reply:
An ordinary service has a ClusterIP address which implements some rudimentary load-balancing capabilities, and exposes the service through the ClusterIP on apiserver.<namespace>.cluster.local. We use this service for internal communication between microservices. 

The headless service has no ClusterIP address, instead when you query the DNS on apiserver-hl.<namespace>.cluster.local, you get back the list of all the pod IP addresses in the response. This is then used by the API gateway to do some smart load balancing for the ingress.